src/dbnode/network/server/tchannelthrift/node/service.go

```diff
@@ -399,13 +428,40 @@ func (s *service) readDatapoints(
 }
 
 func (s *service) FetchTagged(tctx thrift.Context, req *rpc.FetchTaggedRequest) (*rpc.FetchTaggedResult_, error) {
+	var (
+		ctx context.Context
+		sp  opentracing.Span
+		ok  bool
+	)
+
+	ctx = tchannelthrift.Context(tctx)
+	ctx, sp, ok = ctx.StartTraceSpan(tracepoint.FetchTaggedSpanID)
```

Comment:
same thing

Reply:
Just wanted to leave comment about avoiding the string casts and varargs slice alloc for the call to `LogFields(...)` when trace is not being sampled.

Reply:
In low frequency methods it probably doesn't matter all that much, but we may as well establish the pattern so that in high frequency hot methods we're avoiding allocs by default with the pattern we establish.